apps/opik-backend/src/main/java/com/comet/opik/domain/TraceService.java

```diff
@@ -253,4 +256,12 @@ public Mono<Boolean> validateTraceWorkspace(@NonNull String workspaceId, @NonNul
                         .allMatch(trace -> workspaceId.equals(trace.workspaceId()))));
     }
 
+    @Override
+    public Mono<TraceCountResponse> countTracesPerWorkspace() {
+        return template.stream(dao::countTracesPerWorkspace)
+                .collectList()
+                .flatMap(items -> Mono.just(new TraceCountResponse(items)))
```

Comment:
Minor: in general, the convention in this service to favour builders instead of constructors.

Reply:
done